index.src.html

```diff
@@ -769,6 +788,191 @@ <h3 id="try-delivery" algorithm>
   8.  Return "`Failure`".
 </section>
 
+<section>
+  <h2 id="observers">Reporting Observers</h2>
+
+  A <dfn>reporting observer</dfn> observes some types of <a>reports</a> from
+  JavaScript, and is represented in JavaScript by the {{ReportingObserver}}
+  object.
+
+  Not all <a>report</a> <a>types</a> are observable from
+  JavaScript. A <a>type</a> can be specified to be <dfn>observable from
+  JavaScript</dfn>, but is not by default.
+
+  Each <a spec=ecmascript lt=realm>ECMAScript global environment</a> has
+  a <dfn>registered reporting observer list</dfn>, which is an <a>ordered
+  set</a> of <a>reporting observers</a>.
+
+  Any <a>reporting observer</a> that is in a <a>registered reporting observer
+  list</a> is considered <dfn>registered</dfn>.
+
+  Each <a spec=ecmascript lt=realm>ECMAScript global environment</a> has
+  a <dfn>report buffer</dfn>, which is a <a spec=infra>list</a>
+  of <a>reports</a> that have been generated in that <a spec=ecmascript
+  lt=realm>ECMAScript global environment</a>. This list is initially empty, and
+  the reports are stored in the same order in which they are generated.
+
+  Note: The purpose of the <a>report buffer</a> is to allow <a>reporting
+  observers</a> to observe reports that were generated earlier than that
+  observer could be created (via the {{buffered}} option). For example, some
+  reports might be generated during an earlier stage of page loading than when
+  an observer could first be created, or before a JavaScript library is loaded
+  that wishes to observe these reports.
+
+  <h3 id=interface-reporting-observer>Interface {{ReportingObserver}}</h3>
+
+  <pre class="idl">
+interface ReportBody {
+};
+
+interface Report {
+  readonly attribute DOMString type;
+  readonly attribute DOMString url;
+  readonly attribute ReportBody? body;
+};
+
+[Constructor(ReportingObserverCallback callback, optional ReportingObserverOptions options)]
+interface ReportingObserver {
+  void observe();
+  void disconnect();
+  ReportList takeRecords();
+};
+
+callback ReportingObserverCallback = void (sequence&lt;Report> reports, ReportingObserver observer);
+
+dictionary ReportingObserverOptions {
+  sequence&lt;DOMString> types;
+  boolean buffered = true;
```

Comment:
Can we change this to `processAlreadyBufferedReports` (or something in that vein)?  Unless there's a general preference in these specs for shorter names, I like the more descriptive style, especially for boolean options.

Reply:
I'd prefer current name because it allows us to keep consistency between *Observer interfaces — e.g. PerformanceObserver, ReportingObserver, etc.

Reply:
SGTM